File: 0070-climbing-stairs/0070-climbing-stairs.ts
function climbStairs(n: number): number {
    const res: number[] = [];
    
    res[0] = 1;
    res[1] = 2  ;
    
    for (let i = 2; i<n; i++) {
        res[i] = res[i-1] + res[i-2]
    }
    
    return res[n-1]
};

File: 0070-climbing-stairs/0070-climbing-stairs.rs
impl Solution {
    pub fn climb_stairs(n: i32) -> i32 {
        
        let n = n as usize; 
     
        let mut steps = vec![0; n+1 ];
        
        steps[0] =1;
        steps[1] =1;
                        
        for i in 2..n+1 {
            steps[i] = steps[i-1] + steps[i-2]
        }
        
        steps[n]
    }
}